pkg/cli/alpha/internal/update.go

```diff
@@ -0,0 +1,353 @@
+package internal
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"sigs.k8s.io/kubebuilder/v4/pkg/config/store/yaml"
+	"sigs.k8s.io/kubebuilder/v4/pkg/machinery"
+)
+
+// Update contains configuration for the update operation
+type Update struct {
+	// FromVersion specifies which version of Kubebuilder to use for the update.
+	// If empty, the version from the PROJECT file will be used.
+	FromVersion string
+}
+
+// Update performs a complete project update by creating a three-way merge to help users
+// upgrade their Kubebuilder projects. The process creates multiple Git branches:
+// - ancestor: Clean state with old Kubebuilder version scaffolding
+// - current: User's current project state
+// - upgrade: New Kubebuilder version scaffolding
+// - merge: Attempts to merge upgrade changes into current state
+func (opts *Update) Update() error {
+	// Load the PROJECT configuration file to get the current CLI version
+	projectConfigFile := yaml.New(machinery.Filesystem{FS: afero.NewOsFs()})
+	if err := projectConfigFile.LoadFrom(yaml.DefaultPath); err != nil { // TODO: assess if DefaultPath could be renamed to a more self-descriptive name
+		return fmt.Errorf("fail to run command: %w", err)
+	}
+
+	// Determine which Kubebuilder version to use for the update
+	cliVersion := projectConfigFile.Config().GetCliVersion()
+
+	// Allow override of the version from PROJECT file via command line flag
+	if opts.FromVersion != "" {
+		// Ensure version has 'v' prefix for consistency with GitHub releases
+		if !strings.HasPrefix(opts.FromVersion, "v") {
+			opts.FromVersion = "v" + opts.FromVersion
+		}
+		log.Infof("Overriding cliVersion field %s from PROJECT file with --from-version %s", cliVersion, opts.FromVersion)
```

Comment:
We are not changing the content of the PROJECT file so I think the wording Overriding can be misleading

Maybe upgrading from from the version X or something like would be a nicer info log